### DISCUSSION TITLE
SCP fails on certain numpy versions

### DISCUSSION THREAD

**Discussion by Matthew-Wilson-49** [semiautomaticgit/SemiAutomaticClassificationPlugin#386] [Q&A] [answered]
Hello there! I've had trouble getting SCP to function due to what I believe are dependency errors. Installing as instructed led to numpy version 2.0.2 to be installed, which caused a numpy error to manifest.
```
Couldn't load plugin 'SemiAutomaticClassificationPlugin' due to an error when calling its classFactory() method 

ImportError: numpy.core.multiarray failed to import
```
I then tried installing numpy version 1.23.5 specifically, which resolved this error but when I try to download satellite images (hitting run in the download products window), the blue window showing that images are downloading shows up for a split second before disappearing (essentially nothing happens on hitting run). No error messages either. So to resolve this I wanted to inquire about what numpy version I should be using along with any other library versions that are important (i.e. what version of PyTorch and scikit-learn). If you have any idea why run is failing with numpy 1.23.5 as well that would be appreciated. I'm using SCP version 8.5.0 along with QGIS-LTR version 3.34.13-Prizren. Thanks in advance!

**Comment by 33-sl** [comment]
Hi! I also had problems with the dependencies, and I downgraded to numpy version 1.21.6. If you have further problems with the dependencies, I would recommend opening the SCP toolbox - Settings - Debug - Test dependencies. Then you'll see where the issue is. Hope this could help!

**Comment by semiautomaticgit** [comment, marked as answer]
> I then tried installing numpy version 1.23.5 specifically, which resolved this error but when I try to download satellite images (hitting run in the download products window), the blue window showing that images are downloading shows up for a split second before disappearing (essentially nothing happens on hitting run). No error messages either. So to resolve this I wanted to inquire about what numpy version I should be using along with any other library versions that are important (i.e. what version of PyTorch and scikit-learn). If you have any idea why run is failing with numpy 1.23.5 as well that would be appreciated. I'm using SCP version 8.5.0 along with QGIS-LTR version 3.34.13-Prizren. Thanks in advance!

Hello @Matthew-Wilson-49 , Numpy version is not a direct requirement of the SCP, but related to other Python libraries that are installed with QGIS, so manually downgrading Numpy version could cause issues in other libraries (unless downgrading also the other libraries that is quite complicated). 

In case of issues I suggest to uninstall QGIS and the Python libraries (if manually installed), reinstall QGIS, and install only Remotior Sensus with the following command
`pip3 install --upgrade remotior-sensus`

This way the SCP will work, but without PyTorch or scikit-learn (used in Random Forest, SVM and MLP classifications).
